File: android/src/main/java/org/ligi/passandroid/model/pass/PassLocation.java
package org.ligi.passandroid.model.pass;

public class PassLocation {

    private String name;
    public double lat;
    public double lon;

    public String getName(Pass pass) {
        if (name == null) {
            // fallback for passes with locations without description - e.g. AirBerlin
            return pass.getDescription();
        }
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}

File: android/src/main/java/org/ligi/passandroid/model/pass/PassLocation.kt
package org.ligi.passandroid.model.pass

class PassLocation {

    var name: String? = null

    var lat: Double = 0.toDouble()
    var lon: Double = 0.toDouble()

    fun getNameWithFallback(pass: Pass) = if (name.isNullOrBlank()) {
        // fallback for passes with locations without description - e.g. AirBerlin
        pass.description
    } else {
        name
    }

    fun getCommaSeparated() = "$lat,$lon"
}
